### PR TITLE
Added/Blockchain RSA Algorithm

### PR DESCRIPTION
**Pull request by SportingDot1024**
### Describe your change:
I have successfully incorporated RSA encryption and decryption into my project using the cryptography library, a Python package. As you may know, RSA is an algorithm that employs private and public keys for data transmission. By leveraging this library, I was able to streamline the processes of generation and encryption/decryption, thereby significantly enhancing data security.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE_NUMBER".

### CODE REVIEW

File: blockchain/rivest_shamir_adleman.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    private_key = rsa.generate_private_key(
+        public_exponent=65537, key_size=2048, backend=default_backend()
+    )
+
+    public_key = private_key.public_key()
+
+    return private_key, public_key
+
+
+def encrypt(message, public_key):
```
[the remark this answers is not in the document]
Comment (SportingDot1024) [reply blockchain/rivest_shamir_adleman.py]: done

File: blockchain/rivest_shamir_adleman.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+"""
+
+from doctest import testmod
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding, rsa
+
+
+def generate_key_pair():
```
[the remark this answers is not in the document]
Comment (SportingDot1024) [reply blockchain/rivest_shamir_adleman.py]: done

File: blockchain/rivest_shamir_adleman.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        padding.OAEP(
+            mgf=padding.MGF1(algorithm=hashes.SHA256()),
+            algorithm=hashes.SHA256(),
+            label=None,
+        ),
+    )
+    return ciphertext
+
+
+def decrypt(ciphertext, private_key):
```
[the remark this answers is not in the document]
Comment (SportingDot1024) [reply blockchain/rivest_shamir_adleman.py]: done

**Comment by SportingDot1024** [comment]
ERROR blockchain/rivest_shamir_adleman.py - ModuleNotFoundError: No module named 'cryptography.'

It passed all the doctest. Why I'm getting this failed for build/build(pull_request)

**Comment by tianyizheng02** [comment]
1. We already have an implementation of RSA in [ciphers/rsa_cipher.py](https://github.com/TheAlgorithms/Python/blob/ecf21bfc87c1d1cd4730e628279b609151bc6c57/ciphers/rsa_cipher.py). If you wish to contribute, please do the due diligence of checking the codebase to see if an algorithm is already implemented.
2. This file is a how-to demonstration of the `cryptography` library. You did not implement the RSA algorithm yourself.
    ```py
    from cryptography.hazmat.backends import default_backend
    from cryptography.hazmat.primitives import hashes
    from cryptography.hazmat.primitives.asymmetric import padding, rsa
    ```

**Pull request closed without merging**